plugins/catalog/src/alpha/EntityOverviewPage.tsx

```diff
@@ -79,18 +91,30 @@ function buildFilterFn(
 function CardWrapper(props: {
   entity: Entity;
   element: React.JSX.Element;
-  filterFunction?: (entity: Entity) => boolean;
+  filterFunction?: (
+    entity: Entity,
+    context: { apis: ApiHolder },
+  ) => boolean | Promise<boolean>;
   filterExpression?: string;
+  fullWidth?: boolean;
 }) {
-  const { entity, element, filterFunction, filterExpression } = props;
+  const { entity, element, filterFunction, filterExpression, fullWidth } =
+    props;
 
   const filterFn = useMemo(
     () => buildFilterFn(filterFunction, filterExpression),
     [filterFunction, filterExpression],
   );
 
-  return filterFn(entity) ? (
-    <Grid item md={6} xs={12}>
+  const apis = useApiHolder();
+
+  const filterResult = useAsync(
+    async () => await filterFn(entity, { apis }),
+    [filterFn, entity, apis],
+  );
+
+  return filterResult.value ? (
+    <Grid item md={fullWidth ? 12 : 6} xs={12}>
```

Comment:
Open to suggestions on `md={fullWidth ? 12 : 6}` 😅

Reply:
In the new catalog pages, we strictly envision having NO columns at all, right? All cards will stack just vertically no matter what.

Then maybe amending that with adding a second fixed column on the side, with its own input for populating with mini-cards.

Like a 9+3 widths layout or so. But targeted with individual inputs, so all cards are always fullwidth

Reply:
Something like this:
![image](https://github.com/backstage/backstage/assets/6290749/d7104cee-4634-4598-bc4f-9459529ac82e)
Adopters can then choose whether the card is in the primary (default) or secondary area? In this case how would you recommend defining this `area` option/config (eg. `'primary' | 'secondary`,  string, or a boolean)?

Reply:
Revert, not needed anymore.